File: rust/src/models/plan_add_on_input.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{add_on_id::AddOnId, price_id::PriceId};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct PlanAddOnInput {
    pub add_on_id: AddOnId,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub max_instances: Option<i32>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub price_id: Option<PriceId>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub self_serviceable: Option<bool>,
}

impl PlanAddOnInput {
    pub fn new(add_on_id: AddOnId) -> Self {
        Self {
            add_on_id,
            max_instances: None,
            price_id: None,
            self_serviceable: None,
        }
    }
}

File: typescript/src/models/planAddOnInput.ts
// this file is @generated
import { type AddOnId, AddOnIdSerializer } from "./addOnId";
import { type PriceId, PriceIdSerializer } from "./priceId";

export interface PlanAddOnInput {
  addOnId: AddOnId;

  maxInstances?: number | null;

  priceId?: PriceId | null;

  selfServiceable?: boolean | null;
}

export const PlanAddOnInputSerializer = {
  _fromJsonObject(object: any): PlanAddOnInput {
    return {
      addOnId: AddOnIdSerializer._fromJsonObject(object["add_on_id"]),
      maxInstances: object["max_instances"],
      priceId:
        object["price_id"] != null
          ? PriceIdSerializer._fromJsonObject(object["price_id"])
          : undefined,
      selfServiceable: object["self_serviceable"],
    };
  },

  _toJsonObject(self: PlanAddOnInput): any {
    return {
      add_on_id: AddOnIdSerializer._toJsonObject(self.addOnId),
      max_instances: self.maxInstances,
      price_id:
        self.priceId != null ? PriceIdSerializer._toJsonObject(self.priceId) : undefined,
      self_serviceable: self.selfServiceable,
    };
  },
};
